x/posts/internal/keeper/keeper.go

```diff
@@ -114,7 +96,7 @@ func (k Keeper) GetPostsFiltered(ctx sdk.Context, params types.QueryPostsParams)
 		matchParentID, matchCreationTime, matchAllowsComments, matchSubspace, matchCreator, matchHashtags := true, true, true, true, true, true
 
 		// match parent id if valid
-		if params.ParentID != nil {
+		if params.ParentID != "" {
```

Comment:
I think it's better to undo this change as a user might want to search for all the posts that do not have a parent (and so `params.ParentID == ""`) . 

There's always a difference between a `params.ParentID == nil` (if a user has not specified any filtering) and `params.ParentID = ""` (if a user wants to search all posts without a parent).